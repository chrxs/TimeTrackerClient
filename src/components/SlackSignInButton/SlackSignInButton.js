import React from 'react'
import PropTypes from 'prop-types'

import styles from './SlackSignInButton.scss'

const SlackSignInButton = ({
  disabled,
  onClick,
  ...otherProps
}) => {
  return (
    <div
      className={styles.SlackSignInButton}
      disabled={disabled}
      onClick={onClick}
      {...otherProps}
    />
  )
}

SlackSignInButton.displayName = 'SlackSignInButton'

SlackSignInButton.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func
}

SlackSignInButton.defaultProps = {
  disabled: false,
  onClick: () => {}
}

export default SlackSignInButton
