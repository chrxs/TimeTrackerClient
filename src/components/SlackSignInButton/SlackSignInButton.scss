@import '../../variables';

.SlackSignInButton {
  background-color: $brand-primary-color;
  border-radius: 6px;
  cursor: pointer;
  position: relative;
  height: 40px;
  width: 172px;

  &:after {
    background: #fff url('./images/sign_in_with_slack@2x.png') 50% 50% no-repeat;
    background-size: contain;
    position: absolute;
    content: '';
    display: block;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 1;
  }

  &:hover,
  &:focus {
    box-shadow: 0 0 3px rgba(0,0,0,.3);
  }

  &:active {
    box-shadow: none;
    &:after {
      opacity: .7;
    }
  }

  &[disabled] {
    background-color: #ccc;
    box-shadow: none;
    cursor: disabled;
    &:after {
      opacity: .7;
    }
  }
}
