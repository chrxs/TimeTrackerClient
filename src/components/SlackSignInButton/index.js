import SlackSignInButton from './SlackSignInButton'

export default SlackSignInButton
