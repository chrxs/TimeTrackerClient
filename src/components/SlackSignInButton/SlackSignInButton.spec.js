/* global describe, test, expect */
import React from 'react'
import SlackSignInButton from './SlackSignInButton'
import renderer from 'react-test-renderer'

describe('SlackSignInButton snapshots', () => {
  test('with just label prop', () => {
    const component = renderer.create(<SlackSignInButton />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with all props', () => {
    const component = renderer.create(<SlackSignInButton disabled={false} onClick={() => {}} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
