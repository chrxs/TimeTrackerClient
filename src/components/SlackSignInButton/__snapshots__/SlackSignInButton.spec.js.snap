// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlackSignInButton snapshots with all props 1`] = `
<div
  className="SlackSignInButton"
  disabled={false}
  onClick={[Function]}
/>
`;

exports[`SlackSignInButton snapshots with just label prop 1`] = `
<div
  className="SlackSignInButton"
  disabled={false}
  onClick={[Function]}
/>
`;
